### PR TITLE
Migrate Elyra extensions to support JupyterLab 4.x

### LINKED ISSUE #3137: Add support for JupyterLab 4.x
**Is your feature request related to a problem? Please describe.**
[JupyterLab 4.0 is now available as a beta](https://github.com/jupyterlab/jupyterlab/releases/tag/v4.0.0b0). We need to evaluate that release and identify what it takes to leverage that release.

- [ ] installation documentation and repository readme https://github.com/elyra-ai/elyra/pull/3145

Resources:
 - [JupyterLab extension v3 -> v4 migration guide](https://jupyterlab.readthedocs.io/en/latest/extension/extension_migration.html)
 - [Elyra issues tagged with `external:jupyterlab 4.0`](https://github.com/elyra-ai/elyra/labels/external%3Ajupyterlab%204.0)

Extensions:

- [ ] Jupyterlab-lsp - current version of jupyterlab-lsp python package depends on jupyterlab being <v4-alpha, v5 release is required for JL 4.0 support [tracker](https://github.com/jupyter-lsp/jupyterlab-lsp/issues/917)

### PR DESCRIPTION
**Pull request by lresende**
### What changes were proposed in this pull request?
Migrate Elyra extensions to support JupyterLab 4.x

Fixes #3137 

### How was this pull request tested?
Building and running on JupyterLab 4.x
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Review by kevin-bates — APPROVED** [review]
I'm assuming the build failures are a temporary hurdle. Thanks for dealing with the @lresende!

**Comment by lresende** [comment]
I have made some progress here, but I would appreciate if I could get some help on the UI pieces, in particular on UI-components... @marthacryan @karlaspuldaro @ajbozarth and others feel free to send prs to this branch as time permits... I completely understand if you are not able to.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -41,7 +41,12 @@ jobs:
             /home/runner/.cache/Cypress
           key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
       - name: Install
-        run: yarn install --frozen-lockfile && tsc -v
+        run: |
+          corepack prepare yarn@3.5.0 --activate
```
Comment (karlaspuldaro) [inline .github/workflows/build.yml:45]: This line is causing build failure, maybe it's best to use a stable version of yarn

**Comment by hadesto** [comment]
hello! is this pull request still being worked on? it would be amazing to have support for JL4 now that JL3 has an official sunset date. Is there any support required from the community? Thank you!

**Comment by hadesto** [comment]
thanks for the quick response!

**Comment by ederign** [comment]
hi @lresende, thank you so much for this work! Where are these 'remaining items' located? I would love to figure out ways to help! (at least testing it)

**Comment by harshad16** [comment]
> thank you so much for this work! Where are these 'remaining items' located? I would love to figure out ways to help! (at least testing it)

@ederign they are opened as PR to the branch, here: https://github.com/lresende/elyra/pull/1

**Comment by Ekzz0** [comment]
Hi @lresende, this migration still being worked on? should we expect changes in the near future?

**Comment by lresende** [comment]
@Ekzz0 we need to resolve some of these conflicts, but, at a high level, the branch is working with JupyterLab 4.2.5

**Comment by caponetto** [comment]
@lresende I've sent two PRs targeting your branch to make it green on the CI. 

Please take a look at them when you have some time.

- https://github.com/lresende/elyra/pull/3
- https://github.com/lresende/elyra/pull/4

I think we should make the CI for this PR green first, and then work on details (for instance reverting those unrelated changes and/or fixing bugs).

**Comment by shalberd** [comment]
Ah, yes, that is the failed yarn cypress integration test again, the one that is flaky.

**Comment by lresende** [comment]
@caponetto It seems like the last test issue is related to some properties not being saved on the pipeline files, which fail to match the existing snapshot. I noticed you made some changes in that area recently. Would you have some time to help identify what's going on?

**Comment by caponetto** [comment]
@lresende please see https://github.com/lresende/elyra/pull/5

**Comment by lresende** [comment]
Thank you @caponetto , so are you saying that the empty fields are not being saved to pipelines anymore and it's ok to remove them?

**Comment by lresende** [comment]
> Thank you @caponetto , so are you saying that the empty fields are not being saved to pipelines anymore and it's ok to remove them? Particularly as it is working on main branch.

Never mind, I see the difference in the duplicate/different block.

**Pull request merged**